### PR TITLE
Remove the skip mark for test_delta_func

### PR DESCRIPTION
**Pull request by njzjz**
## Summary

https://github.com/scipy/scipy/issues/21052 has been fixed and released in scipy 1.14.1. This PR removes the skip mark.

## Todos

If this is work in progress, what else needs to be done?

- feature 2: ...
- fix 2:

## Checklist

- [ ] Google format doc strings added. Check with `ruff`.
- [ ] Type annotations included. Check with `mypy`.
- [ ] Tests added for new features/fixes.
- [ ] If applicable, new classes/functions/modules have [`duecredit`](https://github.com/duecredit/duecredit) `@due.dcite` decorators to reference relevant papers by DOI ([example](https://github.com/materialsproject/pymatgen/blob/91dbe6ee9ed01d781a9388bf147648e20c6d58e0/pymatgen/core/lattice.py#L1168-L1172))

Tip: Install `pre-commit` hooks to auto-check types and linting before every commit:

```sh
pip install -U pre-commit
pre-commit install
```

### CODE REVIEW

**Comment by njzjz** [comment]
All windows tests have passed.

**Comment by DanielYang59** [comment]
Thanks for keeping an eye on this, perhaps we should bumpy `scipy` version also (to `>=1.14.1`)?

https://github.com/materialsproject/pymatgen/blob/41e4c69911f2b724734e0917c436ee5a1de63091/pyproject.toml#L68

**Comment by njzjz** [comment]
> perhaps we should bump `scipy` version also (to `>=1.14.1`)?

I bump it for Windows only.

File: pyproject.toml [hunk 1]
```diff
@@ -66,6 +66,7 @@ dependencies = [
     "requests>=2.32",
     "ruamel.yaml>=0.17.0",
     "scipy>=1.13.0",
+    "scipy>=1.14.1; platform_system == 'Windows'",
```
Comment (janosh) [inline pyproject.toml:69]: @njzjz can you add a comment linking to https://github.com/scipy/scipy/issues/21052 to explain the platform-dependent version pin?

**Review by janosh — CHANGES_REQUESTED** [review]
thanks @njzjz, just a nitpick to address

File: pyproject.toml (hunk 1 above)
Comment (njzjz) [reply pyproject.toml:71]: added in d6a0b5aed6825c969f6370f5b3f9119520d6d20b

**Review by janosh — APPROVED** [review]

**Pull request merged**